core/src/components/datetime/datetime.tsx

```diff
@@ -2011,7 +2035,11 @@ export class Datetime implements ComponentInterface {
 
           <div class="calendar-next-prev">
             <ion-buttons>
-              <ion-button aria-label="Previous month" disabled={prevMonthDisabled} onClick={() => this.prevMonth()}>
+              <ion-button
+                aria-label="Previous month"
+                disabled={prevMonthDisabled || this.disabled}
```

Comment:
Referencing `this` variables incur a slight performance cost in Stencil, so if we want to use the variable multiple times we typically destructure `this` and reference the local variable instead to avoid duplicating the perf cost.

To fix this, you can do the following:

1. Destructure `disabled` at the top of `renderCalendarHeader`: `const { disabled } = this;`
2. Replace `this.disabled` with `disabled` in the function (to reference the local var instead).